server/sonar-process-monitor/src/main/java/org/sonar/application/cluster/ClusterProcess.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.application.cluster;
+
+import java.io.Serializable;
+import javax.annotation.Nonnull;
+import org.sonar.process.ProcessId;
+
+public class ClusterProcess implements Serializable {
+  private final ProcessId processId;
+  private final String nodeUuid;
+
+  public ClusterProcess(@Nonnull String nodeUuid, @Nonnull ProcessId processId) {
+    this.processId = processId;
+    this.nodeUuid = nodeUuid;
+  }
+
+  public ProcessId getProcessId() {
+    return processId;
+  }
+
+  public String getNodeUuid() {
+    return nodeUuid;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof ClusterProcess)) {
+      return false;
+    }
+
+    ClusterProcess that = (ClusterProcess) o;
+
+    if (processId != that.processId) {
+      return false;
+    }
+    return nodeUuid != null ? nodeUuid.equals(that.nodeUuid) : that.nodeUuid == null;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Add parentheses to make the operator precedence explicit. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS864)